### ISSUE TITLE
[CT-540] [Bug] 'target' suddenly not available at compile time in selectors.yml

### ISSUE THREAD

**Issue by ljhopkins2** [dbt-labs/dbt-core#5131]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current Behavior

After upgrading to `dbt v1.0.5`, I get the following error when compiling my model:
```
Compilation Error
  Could not render {{ target.name == 'dev' | as_bool }}: 'target' is undefined
```

### Expected Behavior

The project should compile without error, as it did in earlier post-v1 releases.

### Steps To Reproduce

1. Add a selector like this to `selectors.yml`
```yml
selectors:
  - name: dev_defer_snapshots
    default: "{{ target.name == 'dev' | as_bool }}"
    definition:
      method: fqn
      value: '*'
      exclude:
        - method: config.materialized
          value: snapshot
```
2. Compile the project with `dbt v1.0.5`
3. Note the error
4. Comment out the line containing `default: "{{ target.name == 'dev' | as_bool }}"`
5. Compile the project
6. Note no error

### Environment

```markdown
- OS: MacOS 12.3.1
- Python: 3.9.10
- dbt: 1.0.5
```

### What database are you using dbt with?

redshift

**Comment by jtcohen6** [comment]
@ljhopkins2 Thanks for the report! We'll get on this.

Could it be an unintended side-effect of https://github.com/dbt-labs/dbt-core/pull/4878?

The `target`-based `default` use case is an increasingly common one. In addition to the fix here, we need a test case for it.

**Comment by jtcohen6** [comment]
Confirmed that change looks to be the cause. I think `selectors.yml` + `packages.yml` use the same rendering context today. We may want to split those apart. We've talked about wanting to support `target` in the former but not the latter, since `target` is not set during `deps`.

@ljhopkins2 In the meantime, could you work around this by using an `env_var` instead of `target.name`?

**Comment by ljhopkins2** [comment]
@jtcohen6 Thanks for suggesting a workaround. I'll probably just leave it commented out for now, though: short-term I can just remember to `--exclude` in the dev env; the bigger impact for me was that it was causing cloud jobs to fail, and that was easy to work around with a `#`!

**Comment by gshank** [comment]
Using a target context for the selector renderer was introduced in 1.0.0.  The secret renderers were passed a context dictionary instead of a cli_var dictionary, so they were broken for var usage. In 1.0.5 we fixed the secret renders so that they were passed a cli_var dictionary instead of the context dictionary, but over-corrected and changed the selector renderer so that it also was just passed a cli_var dictionary. This PR changes the selector rendering to use the DbtProjectYamlRenderer, since the 'render_selectors' call is contained within it and needs the same rendering setup that it has.

**Comment by saraleon1** [comment]
+1 dbt Cloud enterprise customer. This is resulting in a failure to all jobs due to a failed dbt deps due to a 'target undefined' error

**Comment by jtcohen6** [comment]
@ljhopkins2 @saraleon1 This fix is now available in v1.0.6-rc1 and v1.1.0-rc3. Those are available in dbt Cloud, respectively, as `1.0 (patch prerelease)` and `1.1 (minor prerelease)`.

Thank you so much for reporting the regression, and helping us get to the bottom of it!
